File: 485. Max Consecutive Ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        cnt, maxcnt = 0, 0 # maxcnt save the max consecutive1s
        for i in nums:
            cnt += 1
            cnt *= i # if i = 0 then cnt = 0
            maxcnt = max(cnt, maxcnt) # update maxcnt
        return max(cnt, maxcnt) #incase the last is longest cons1s.

File: 485. Max Consecutive Ones/485. Max Consecutive Ones.c
int findMaxConsecutiveOnes(int* nums, int numsSize) {
    int cnt = 0, maxcnt = 0, i = 0;
    for (i = 0; i < numsSize; i++)
        if (nums[i] == 1)
            cnt++;
        else{
            maxcnt = (maxcnt > cnt ? maxcnt : cnt);
            cnt = 0;
        }
    return maxcnt > cnt ? maxcnt : cnt;
}
